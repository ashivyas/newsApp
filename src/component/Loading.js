import React, { Component } from "react";
import { Content, Spinner, Text } from "native-base";

export default class Loading extends Component {
  render() {
    return (
      <Content contentContainerStyle={styles.mainContentContainerStyle}>
        <Spinner color={"#F00"} />
        <Text style={styles.msgStyle}>Please wait...</Text>
      </Content>
    );
  }
}

let styles = {
  msgStyle: {
    color: "#F00",
    fontFamily: "gsr"
  },
  mainContentContainerStyle: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  }
};
