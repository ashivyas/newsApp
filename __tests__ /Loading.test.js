import React from "react";
import Loading from "../src/component/Loading";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
