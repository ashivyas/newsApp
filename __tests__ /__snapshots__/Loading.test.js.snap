// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  viewIsInsideTabBar={false}
>
  <View>
    <ActivityIndicator
      animating={true}
      color="#F00"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "height": 80,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#F00",
            "fontFamily": "gsr",
          },
        ]
      }
      uppercase={false}
    >
      Please wait...
    </Text>
  </View>
</RCTScrollView>
`;
